### PR TITLE
MODOAIPMH-513 - Avoid skipping entire batch when 500 Internal Server Error from inventory

### PR DESCRIPTION
**Pull request by alekGbuz**
[MODOAIPMH-513](https://issues.folio.org/browse/MODOAIPMH-513) - Avoid skipping entire batch when 500 Internal Server Error from inventory

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] Check logging
  - [ x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by alekGbuz** [comment]
karate-mod-oai-pmh

**Comment by alekGbuz** [comment]
karate-edge-oai-pmh

File: src/main/java/org/folio/oaipmh/processors/OaiPmhJsonParser.java [hunk 1]
```diff
@@ -54,4 +59,8 @@ public void end() {
       throw e;
     }
   }
+
+  public List<String> getErrors() {
+    return errors;
```
Comment (obozhko-folio) [inline src/main/java/org/folio/oaipmh/processors/OaiPmhJsonParser.java:64]: If this list of errors is not supposed to be changed, wouldn't be better to return immutable list here?

File: src/main/java/org/folio/oaipmh/helpers/enrichment/ItemsHoldingsRequestWithDelayExecutor.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import static org.folio.oaipmh.Constants.SKIP_SUPPRESSED_FROM_DISCOVERY_RECORDS;
+import static org.folio.oaipmh.helpers.AbstractGetRecordsHelper.INSTANCE_IDS_ENRICH_PARAM_NAME;
+
+public class ItemsHoldingsRequestWithDelayExecutor {
+
+  private static final Logger logger = LogManager.getLogger(ItemsHoldingsRequestWithDelayExecutor.class);
+
+  private final Vertx vertx = Vertx.currentContext().owner();
+  private final ItemsHoldingsEnrichment itemsHoldingsEnrichment;
+  private final AtomicInteger inc = new AtomicInteger(0);
```
Comment (obozhko-folio) [inline src/main/java/org/folio/oaipmh/helpers/enrichment/ItemsHoldingsRequestWithDelayExecutor.java:25]: Default value in AtomicInteger is 0 anyway and, probably, it's not necessary to indicate it explicitly in constructor.

**Review by obozhko-folio — APPROVED** [review]

**Review by firdavs-m — APPROVED** [review]

File: src/main/java/org/folio/oaipmh/helpers/enrichment/ItemsHoldingsRequestWithDelayExecutor.java (hunk 2 above)
Comment (alekGbuz) [reply src/main/java/org/folio/oaipmh/helpers/enrichment/ItemsHoldingsRequestWithDelayExecutor.java]: Thanks, fixed.

File: src/main/java/org/folio/oaipmh/processors/OaiPmhJsonParser.java (hunk 1 above)
Comment (alekGbuz) [reply src/main/java/org/folio/oaipmh/processors/OaiPmhJsonParser.java:64]: @obozhko-folio , I think in that case it is possible to use mutable list as well as there is no any logic to change errors list.

**Pull request merged**